apis/kueue/v1beta1/workload_types.go

```diff
@@ -334,6 +334,17 @@ const (
 	WorkloadDeactivationTarget = "DeactivationTarget"
 )
 
+// Reasons for the WorkloadPreempted condition.
+const (
+	// In cluster queue preemption reasons
+	InClusterQueueReason string = "InClusterQueue"
```

Comment:
```suggestion
	// InClusterQueueReason indicates the preemption reason
	// when the Workload is preempted by the preemptee within the same ClusterQueue.
	InClusterQueueReason string = "InClusterQueue"
```

Could you add a more granular comment since this is exposed as APIs and the API documentation are exposed?

Additionally, could you address the other 3 preemption reasons as well?

Reply:
Good point, I have for now updated based on the https://github.com/kubernetes-sigs/kueue/blob/e38b145f99ec05bea96679423239a22d920e1487/pkg/scheduler/preemption/preemption.go#L184-L189. Let me know if they should be more descriptive than that.